### PR TITLE
docs(readme): add quickstart guide

### PR DESCRIPTION
**Pull request by sammyne**
Add a quickstart guide in README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
...
+    cargo build --release -p hello-component-client --target wasm32-wasi
+    ```
+    > Output is target/wasm32-wasi/release/hello-component-client.wasm.
+1. Run NATS service using docker
+    ```bash
+    docker run -it --name nats-server -p 4222:4222 nats:2.10.20-alpine3.20
+    ```
+1. Serve the server WASM component
+    ```bash
+    cargo run --release --bin wrpc-wasmtime-nats -- serve -g hello-world hello world ./target/wasm32-wasi/release/hello_component_server.wasm
```
Comment (rvolosatovs) [inline README.md:71]:
```suggestion
    cargo run --release --bin wrpc-wasmtime-nats -- serve -g hello rust rust ./target/wasm32-wasi/release/hello_component_server.wasm
```

Would be more aligned with other examples (currently, native Go examples listen on `go` prefix and native Rust examples listen on `rust` prefix by default). `hello` is the name of the example app

File: README.md [hunk 2]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
...
+    ```
+1. Call the server WASM component from the client WASM component
+    ```bash
+    cargo run --release --bin wrpc-wasmtime-nats -- run world ./target/wasm32-wasi/release/hello-component-client.wasm
```
Comment (rvolosatovs) [inline README.md:75]:
```suggestion
    cargo run --release --bin wrpc-wasmtime-nats -- run rust ./target/wasm32-wasi/release/hello-component-client.wasm
```

File: README.md [hunk 3]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
...
+[examples/rust/hello-component-client](examples/rust/hello-component-client).
+
+### Requirement
+- docker >= 24.0.6, build ed223bc
+- rust >= 1.80.1
+
+### How-To
+1. Add the `wasm32-wasi` target
+    ```bash
+    rustup target add wasm32-wasi
```
Comment (rvolosatovs) [inline README.md:53]: We should probably use `wasm32-wasip1`, which is the standard name for legacy WASI version https://blog.rust-lang.org/2024/04/09/updates-to-rusts-wasi-targets.html

I'd prefer `wasm32-wasip2` here, but I think we should stick to `wasm32-wasip1`, which is available in stable

Perhaps we could add a note that `wasm32-wasip2` can also be used?

File: README.md [hunk 4]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
...
+    > Output is target/wasm32-wasi/release/hello_component_server.wasm.
+1. Build the client WASM component
+    ```bash
+    cargo build --release -p hello-component-client --target wasm32-wasi
+    ```
+    > Output is target/wasm32-wasi/release/hello-component-client.wasm.
+1. Run NATS service using docker
+    ```bash
+    docker run -it --name nats-server -p 4222:4222 nats:2.10.20-alpine3.20
+    ```
```
Comment (rvolosatovs) [inline README.md:68]: Could we add an option here to run the `nats-server` binary directly without using Docker?

File: README.md [hunk 5]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
...
 wRPC fully supports the unreleased native [WIT] `stream` and `future` data types along with all currently released WIT functionality.
 
+## Quickstart
+
+Here is a guide demonstrating how to bootstrap the WASM component based example implemented by
+[examples/rust/hello-component-server](examples/rust/hello-component-server) and 
+[examples/rust/hello-component-client](examples/rust/hello-component-client).
+
+### Requirement
+- docker >= 24.0.6, build ed223bc
```
Comment (rvolosatovs) [inline README.md:47]:
```suggestion
- `docker` >= 24.0.6 or `nats-server` >= 2.10.20
```

File: README.md [hunk 6]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
 
 wRPC fully supports the unreleased native [WIT] `stream` and `future` data types along with all currently released WIT functionality.
 
+## Quickstart
+
+Here is a guide demonstrating how to bootstrap the WASM component based example implemented by
```
Comment (rvolosatovs) [inline README.md:42]: nit; the correct casing is actually *Wasm*
```suggestion
Here is a guide demonstrating how to bootstrap the Wasm component based example implemented by
```

File: README.md [hunk 7]
```diff
@@ -37,6 +37,49 @@ For static use cases, wRPC provides [WIT] binding generators for:
 
 wRPC fully supports the unreleased native [WIT] `stream` and `future` data types along with all currently released WIT functionality.
 
+## Quickstart
+
```
Comment (rvolosatovs) [inline README.md:41]: Could we add a small note here that there are 2 different kinds of examples - native wRPC applications, using a particular wRPC transport (currently, NATS only) and *generic* Wasm components, that run in a Wasm runtime?

**Review by rvolosatovs — APPROVED** [review]
Thanks!

Happy to merge as-is and address my own feedback in a follow-up, let me know what you prefer!
Generally, I'm a bit concerned about giving the wrong impression that an OCI runtime (like Docker) or some external service is required to use wRPC. It's true that NATS is the only transport currently supported for *both* Go and Rust, but it's certainly not a requirement.

**Comment by sammyne** [comment]
Updated as suggested. Check it~@rvolosatovs

**Pull request merged**